system/lib/libc/dynlink.c

```diff
@@ -169,6 +181,27 @@ static struct dso* load_library_start(const char* name, int flags) {
   p->flags = flags;
   strcpy(p->name, name);
 
+  // If the file exists in the filesystem, load it here into linear memory which
+  // makes the data available to JS, and to other threads.  This data gets
+  // free'd later once all threads have loaded the DSO.
+  struct stat statbuf;
+  if (stat(name, &statbuf) == 0 && S_ISREG(statbuf.st_mode)) {
+    int fd = open(name, O_RDONLY);
+    if (fd >= 0) {
+      off_t size = lseek(fd, 0, SEEK_END);
+      if (size != (off_t)-1) {
+        lseek(fd, 0, SEEK_SET);
+        p->file_data = malloc(size);
```

Comment:
Should we check for a failing `malloc`? (I don't think we can depend on `p->file_data` being null leading to an error in `read`)